pkg/query-service/app/integrations/messagingQueues/kafka/translator.go

```diff
@@ -0,0 +1,104 @@
+package kafka
+
+import (
+	"fmt"
+
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
+)
+
+var defaultStepInterval int64 = 60
+
+func BuildQueryRangeParams(messagingQueue *MessagingQueue, queryContext string) (*v3.QueryRangeParamsV3, error) {
+
+	// ToDo: propagate this through APIs when there are different handlers
+	queueType := kafkaQueue
+
+	var cq *v3.CompositeQuery
+	if queryContext == "producer" {
+		chq, err := buildProducerClickHouseQuery(messagingQueue, queueType)
+		if err != nil {
+			return nil, err
+		}
+		cq, err = buildCompositeQueryProducer(chq)
+
+		if err != nil {
+			return nil, err
+		}
+	} else if queryContext == "consumer" {
+		chq, err := buildConsumerClickHouseQuery(messagingQueue, queueType)
+		if err != nil {
+			return nil, err
+		}
+		cq, err = buildCompositeQueryConsumer(chq)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	queryRangeParams := &v3.QueryRangeParamsV3{
+		Start:          messagingQueue.Start,
+		End:            messagingQueue.End,
+		Step:           defaultStepInterval,
+		CompositeQuery: cq,
+		Version:        "v4",
+		FormatForWeb:   true,
+	}
+
+	return queryRangeParams, nil
+}
+
+func buildProducerClickHouseQuery(messagingQueue *MessagingQueue, queueType string) (*v3.ClickHouseQuery, error) {
+	start := messagingQueue.Start
+	end := messagingQueue.End
+	topic, ok := messagingQueue.Variables["topic"]
+	if !ok {
+		return nil, fmt.Errorf("invalid type for Topic")
+	}
+
+	partition, ok := messagingQueue.Variables["partition"]
+
+	if !ok {
+		return nil, fmt.Errorf("invalid type for Partition")
+	}
```

Comment:
The extraction of what you need from the messagingQueue struct for your clickhouse query should be moved to a common function.

Right now, it's duplicated in `buildProducerClickHouseQuery` and  `buildConsumerClickHouseQuery`. The more functions you add, the more difficult it starts to get to maintain.

Reply:
True, created in haste, jumbling with many things, let me fix this

Reply:
fixed